File: core/src/com/xenoage/zong/core/music/clef/Clef.java
package com.xenoage.zong.core.music.clef;

import lombok.Data;
import lombok.EqualsAndHashCode;

import com.xenoage.utils.annotations.NonNull;
import com.xenoage.zong.core.music.MeasureElement;
import com.xenoage.zong.core.position.MPContainer;


/**
 * Class for a clef.
 * 
 * @author Andreas Wenger
 */
@Data @EqualsAndHashCode(exclude="parent")
public final class Clef
	implements MeasureElement
{

	/** The type of the clef. */
	@NonNull private ClefType type;
	
	/** Back reference: the parent element, or null, if not part of a score. */
	private MPContainer parent;

}

File: core/src/com/xenoage/zong/core/music/clef/Clef.kt
package com.xenoage.zong.core.music.clef

import com.xenoage.zong.core.music.Measure
import com.xenoage.zong.core.music.MeasureElement


/**
 * Class for a clef.
 */
class Clef(
		/** The type of the clef. */
		val type: ClefType
) : MeasureElement {

	override var parentMeasure: Measure? = null

}
